src/probe/probe.c

```diff
@@ -1099,23 +1152,69 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 		_probe->probe_points[first_free].stream_tag =
 			probe[i].stream_tag;
 
-		if (probe[i].purpose == PROBE_PURPOSE_LOGGING) {
+		if (probe[i].purpose == PROBE_PURPOSE_LOGGING || probe[i].buffer_id.fields.module_id == 0) {
 #if CONFIG_LOG_BACKEND_SOF_PROBE
 			probe_logging_init(probe_logging_hook);
 #else
 			return -EINVAL;
 #endif
 		} else {
-			notifier_register(_probe, dev->cb, NOTIFIER_ID_BUFFER_PRODUCE,
+#if CONFIG_IPC_MAJOR_4
+			notifier_register(&probe[i].buffer_id.full_id, buf, NOTIFIER_ID_BUFFER_PRODUCE,
+					  &probe_cb_produce, 0);
+			notifier_register(&probe[i].buffer_id.full_id, buf, NOTIFIER_ID_BUFFER_FREE,
+					  &probe_cb_free, 0);
+#else
+			notifier_register(&probe[i].buffer_id.full_id, dev->cb, NOTIFIER_ID_BUFFER_PRODUCE,
```

Comment:
Hmm, this doesn't seem right, code is identical to IPC4 branch. "buffer_id.full_id" is in the IPC4 ABI only, right?

Reply:
Yeap, that was my plan to unify this and use full_id everywhere so we dont have to branch for IPC all the time but forget to update ipc3 header which is done now :)